engine/apps/metrics_exporter/helpers.py

```diff
@@ -0,0 +1,227 @@
+
+    instance_slug = integration.organization.stack_slug
+    instance_id = integration.organization.stack_id
+    grafana_org_id = integration.organization.org_id
+    metric_alert_groups_total: typing.Dict[int, AlertGroupsTotalMetricsDict] = cache.get(
+        metric_alert_groups_total_key, {}
+    )
+    metric_alert_groups_total.setdefault(
+        integration.id,
+        {
+            "integration_name": integration.emojized_verbal_name,
+            "team_name": integration.team_name,
+            "team_id": integration.team_id_or_no_team,
+            "org_id": grafana_org_id,
+            "slug": instance_slug,
+            "id": instance_id,
+            STATE_FIRING: 0,
+            STATE_ACKNOWLEDGED: 0,
+            STATE_RESOLVED: 0,
+            STATE_SILENCED: 0,
+        },
+    )
+    cache.set(metric_alert_groups_total_key, metric_alert_groups_total, timeout=metrics_cache_timeout)
+
+    metric_alert_groups_response_time_key = get_metric_alert_groups_response_time_key(integration.organization_id)
+    metric_alert_groups_response_time: typing.Dict[int, AlertGroupsResponseTimeMetricsDict] = cache.get(
+        metric_alert_groups_response_time_key, {}
+    )
+    metric_alert_groups_response_time.setdefault(
+        integration.id,
+        {
+            "integration_name": integration.emojized_verbal_name,
+            "team_name": integration.team_name,
+            "team_id": integration.team_id_or_no_team,
+            "org_id": grafana_org_id,
+            "slug": instance_slug,
+            "id": instance_id,
+            "response_time": [],
+        },
+    )
+    cache.set(metric_alert_groups_response_time_key, metric_alert_groups_response_time, timeout=metrics_cache_timeout)
+
+
+def metrics_bulk_update_team_label_cache(teams_updated_data, organization_id):
+    """Update team related data in metrics cache for each team in `teams_updated_data`"""
+    if not teams_updated_data:
+        return
+    metrics_cache_timeout = get_metrics_cache_timeout(organization_id)
+    metric_alert_groups_total_key = get_metric_alert_groups_total_key(organization_id)
+    metric_alert_groups_response_time_key = get_metric_alert_groups_response_time_key(organization_id)
+
+    metric_alert_groups_total = cache.get(metric_alert_groups_total_key, {})
+    metric_alert_groups_response_time = cache.get(metric_alert_groups_response_time_key, {})
+    for team_id, team_data in teams_updated_data.items():
+        for integration_id in metric_alert_groups_total:
+            if metric_alert_groups_total[integration_id]["team_id"] == team_id:
+                integration_response_time_metrics = metric_alert_groups_response_time.get(integration_id)
+                if team_data["deleted"]:
+                    metric_alert_groups_total[integration_id]["team_id"] = "no_team"
+                    metric_alert_groups_total[integration_id]["team_name"] = "No team"
+                    if integration_response_time_metrics:
```

Comment:
nit: Maybe it's possible to simplify this a bit? It's hard to keep context when there are 4 levels of indentation.

Reply:
Checking, not seeing an easy refactoring atm, but for sure maybe something to revisit later when we deal with more metrics.